File: my_cpp_practice/src/add_two_ints_client_no_oop.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<rclcpp::Node>("add_two_ints_client_no_oop");

  auto client = node->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
  while (!client->wait_for_service(std::chrono::seconds(1)))
  {
    RCLCPP_WARN(node->get_logger(), "Waiting for the add_two_ints_server to be up...");
  }

  auto req = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
  req->a = 3;
  req->b = 8;

  auto future = client->async_send_request(req);
  if (rclcpp::spin_until_future_complete(node, future) == rclcpp::FutureReturnCode::SUCCESS)
  {
    RCLCPP_INFO(node->get_logger(), "%d + %d = %d", (int)req->a, (int)req->b, (int)future.get()->sum);
  }
  else
  {
    RCLCPP_ERROR(node->get_logger(), "Error while calling add_two_ints_service");
  }

  rclcpp::shutdown();
  return 0;
}

File: my_py_practice/my_py_practice/add_two_ints_client_no_oop.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.srv import AddTwoInts


def main(args=None):
    rclpy.init(args=args)
    node = Node("add_two_ints_client_no_oop")

    client_ = node.create_client(AddTwoInts, "add_two_ints")

    while not client_.wait_for_service(timeout_sec=1.0):
        node.get_logger().warn("waithing for add_two_int_server ...")

    req = AddTwoInts.Request()
    req.a = 3
    req.b = 4

    future = client_.call_async(req)
    rclpy.spin_until_future_complete(node, future)

    try:
        res = future.result()
        node.get_logger().info("%ld + %ld = %ld" % (req.a, req.b, res.sum))
    except Exception as e:
        node.get_logger().error("Service call failed %r" % (e,))
    
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()